test/Riok.Mapperly.Tests/Generator/IncrementalGeneratorTest.cs

```diff
@@ -0,0 +1,161 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Riok.Mapperly.Tests.Generator;
+
+[UsesVerify]
+public class IncrementalGeneratorTest
+{
+    [Fact]
+    public void AddingUnrelatedTypeDoesNotRegenerateOriginal()
+    {
+        var source = TestSourceBuilder.CSharp(
+            """
+            using Riok.Mapperly.Abstractions;
+
+            namespace Test.A
+            {
+                [Mapper]
+                internal partial class FooBarMapper
+                {
+                    internal partial string FooToBar(string value);
+                }
+            }
+            """
+        );
+
+        var syntaxTree = CSharpSyntaxTree.ParseText(source, CSharpParseOptions.Default);
+        var compilation1 = TestHelper.BuildCompilation(TestHelperOptions.NoDiagnostics.NullableOption, syntaxTree);
+
+        var driver1 = TestHelper.TrackingGeneratorDriver().RunGenerators(compilation1);
+
+        var compilation2 = compilation1.AddSyntaxTrees(CSharpSyntaxTree.ParseText("struct MyValue {}", CSharpParseOptions.Default));
```

Comment:
I think it would make sense to add a `SyntaxTree` method (and probably additional overloads to `Mapping`: `MappingSyntaxTree`) to the `TestSourceBuilder` to improve the readability of the tests. Eg. `TestSourceBuilder.SyntaxTree("struct MyValue {}")`. This would also improve IDE support as some IDE's recognise the `StringSyntax` `csharp` (eg. Rider), which `ParseText` doesn't have.